internal/provider/pod.go

```diff
@@ -406,32 +406,37 @@ func unitPrefix(namespace, podName string) string {
 	return prefix + separator + namespace + separator + podName
 }
 
-func Image(name string) string {
-	el := strings.Split(name, separator) // assume well formed
+// Name returns <namespace>.<podname> from a well formed name.
```

Comment:
What's the definition of a _well-formed name_? I would rather have this take a namespace, pod and container strings than an opaque value that may or may not contain a valid concatenation of those. By doing so, we'd even get rid of string splitting.